pkgs/development/tools/build-managers/meson/setup-hook.sh

```diff
@@ -25,3 +25,17 @@ if [ -z "$dontUseMesonConfigure" -a -z "$configurePhase" ]; then
     setOutputFlags=
     configurePhase=mesonConfigurePhase
 fi
+
+mesonCheckPhase() {
+    runHook preCheck
+
+    echo "meson check flags: $mesonCheckFlags"
+    meson test $mesonCheckFlags
```

Comment:
Are you sure that `mesonCheckFlags` will be useful?

`meson test` is a more obvious label than `meson check`.

Reply:
It can be used for running tests [in a wrapper](http://mesonbuild.com/Unit-tests.html#testing-tool) (i.e. valgrind). Not sure if it is useful for packages.

Reply:
As for the name, I wanted to be consistent with the rest of the Nix. There is `doCheck` attribute and `checkPhase` is also used in python derivations even though the invocation is `./setup.py test`.

Reply:
`checkPhase` is common to all Nixpkgs, and calling it `testPhase` just because the test command is `… test` would be confusing. However, flags are specific to tools and their variables contain the name of the tool, so passing flags to `meson test` in `$mesonCheckFlags` is surprising. (I'd rather just remove both the variable and the echo.)

Reply:
Removed. In the future we would probably want to use `checkTarget` like triton [does](https://github.com/triton/triton/blob/0e7cdad3be0998a8df82bdd0b76b187a1ff557be/pkgs/all-pkgs/m/meson/setup-hook.sh#L50) but that would require changes to [ninja setup hook](https://github.com/triton/triton/blob/0e7cdad3be0998a8df82bdd0b76b187a1ff557be/pkgs/all-pkgs/n/ninja/setup-hook.sh#L37-L47) as well.